lib/passageidentity/auth.rb

```diff
@@ -3,21 +3,130 @@
 require 'openssl'
 require 'base64'
 require 'jwt'
+require 'rubygems/deprecate'
 require_relative 'client'
 require_relative '../openapi_client'
 
 module Passage
   # The Passage::Auth class provides methods for authenticating requests and tokens
   class Auth
+    extend Gem::Deprecate
+
     # rubocop:disable Metrics/AbcSize
-    def initialize(app_id, auth_strategy)
+    def initialize(app_id, api_key, auth_strategy)
       @app_cache = {}
       @app_id = app_id
+      @api_key = api_key
       @auth_strategy = auth_strategy
 
       fetch_jwks
     end
 
+    def authenticate_request(request)
+      # Get the token based on the strategy
+      if @auth_strategy == Passage::COOKIE_STRATEGY
+        unless request.cookies.key?('psg_auth_token')
+          raise PassageError.new(
+            message:
+              'missing authentication token: expected "psg_auth_token" cookie'
+          )
+        end
+        @token = request.cookies['psg_auth_token']
+      else
+        headers = request.headers
+        raise PassageError.new(message: 'no authentication token in header') unless headers.key?('Authorization')
+
+        @token = headers['Authorization'].split(' ').last
+      end
+
+      validate_jwt(@token)
+    end
+
+    def validate_jwt(token)
+      return authenticate_token(token) if token
+
+      raise PassageError.new(message: 'no authentication token')
+    end
+
+    def revoke_user_refresh_tokens(user_id)
+      warn 'NOTE: Passage::Auth#revoke_user_refresh_tokens is deprecated;
+        use Passage::User#revoke_refresh_tokens instead. It will be removed on or after 2024-12.'
```

Comment:
Leaving the warn and the duplicated logic. In order to access  `Passage::User#revoke_refresh_tokens` I would have to rework the Auth class. I didn't think it was worth reworking the Auth class to later remove all of the changes.